kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt

```diff
@@ -34,9 +34,20 @@ import com.google.android.exoplayer2.upstream.*
 import com.google.android.exoplayer2.util.Util
 import timber.log.Timber
 import java.util.concurrent.TimeUnit
+import com.google.android.exoplayer2.upstream.cache.SimpleCache
+import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor
 
-abstract class BaseAudioPlayer internal constructor(private val context: Context, bufferConfig: BufferConfig? = null) : AudioManager.OnAudioFocusChangeListener {
+import com.google.android.exoplayer2.database.ExoDatabaseProvider
+
+import com.google.android.exoplayer2.database.DatabaseProvider
+import com.google.android.exoplayer2.upstream.cache.CacheDataSource
+import java.io.File
+
+
+abstract class BaseAudioPlayer internal constructor(private val context: Context, bufferConfig: BufferConfig? = null, private val cacheConfig: CacheConfig? = null) : AudioManager.OnAudioFocusChangeListener {
```

Comment:
This is an internal constructor, so `cacheConfig` cannot be passed from the outside. It needs to come from `AudioPlayer` and `QueuedAudioPlayer`'s constructors.